eng/tools/compare-packages.ps1

```diff
@@ -0,0 +1,63 @@
+param (
+  $pathToMasterPkg,
+  $pathToCurrentPkg,
+  $dailyDevBuildNo,
+  $pathForDiffFile
+)
+
+function ExtractTGZPackages($pathToPkg){
+  #$regExp = "-\d+\.\d+\.\d+(-\w*\.\d*)?(-dev-$dailyDevBuildNo)?"
+
+  $parentExtractDir = Join-Path $pathToPkg "all-contents"
+  Write-Host "mkdir $parentExtractDir"
+  mkdir $parentExtractDir
+
+  foreach ($p in $(dir $pathToPkg -r -i *.tgz)){
+    $regExp = "(?<name>.*?)(-\d+\.\d+\.\d+.*)"
+    $extractDir = Join-Path $parentExtractDir $p.BaseName
```

Comment:
You shouldn't need this line.

Reply:
Aren't the powershell variables scoped like other languages? Like if I declare the $extractDir variable within the if block, can i access it outside too?

Reply:
They don't scope at this level only at the script or global level.

Reply:
In that case should i move the other commands - mkdir, pushd,popd within the if statement - because we don't want them to execute if we don't have the correct folder name?

Reply:
If you feel more comfortable initializing this I'd initialize it to empty to avoid us accidentally extracting the contents to another location.

Reply:
Actually I take that back if we do end up with empty then it might extract things somewhere completely different. Instead I suggest moving all the extract work into the if block.

Reply:
done